### ISSUE TITLE
False positive and type-checking error: examples/lib/Test.agda

### ISSUE THREAD

**Issue by asr** [agda/agda#1887]
The above module isn't being tested:

```
$ cd examples
$ make lib 
make: `lib' is up to date.
```

After renaming the `lib` rule to `library` (or adding a .`PHONY` target name) in Makefile, I got the following error:

```
$ make library
Testing lib/Test.agda... 
/agda/examples/lib/Data/Nat/Properties.agda:38,15-19
Type mismatch
when checking that the pattern zero has type _A_0
make: *** [library] Error 1
```

**Comment by asr** [comment]
BTW, I don't know why I can _only_ reproduce this error when testing Agda with GHC 8.0.1 RC 2.

**Comment by asr** [comment]
It seems `examples/lib/Test.agda` hasn't been tested at least from 2010! I'm getting errors related to the fix 42de243.
